src/Nethermind/Nethermind.Trie/NibblePath.cs

```diff
@@ -358,40 +385,85 @@ public Ref Slice(int start)
                 (byte)(Length - start));
         }
 
-        public Ref SliceRef(int currentIndex, int remainingUpdatePathLength)
+        public Ref Slice(int start, int length)
         {
-            throw new NotImplementedException();
+            Debug.Assert(start + length <= Length, "Cannot slice the NibblePath beyond its Length");
+            return new(ref Unsafe.Add(ref _span, (_odd + start) / 2),
+                (byte)((start & 1) ^ _odd), (byte)length);
         }
 
         public string ToHexString()
         {
-            throw new NotImplementedException();
+            const int prefixLength = 2;
+
+            // TODO: optimize
```

Comment:
`String.Create<TState>(Int32, TState, SpanAction<Char,TState>)` https://learn.microsoft.com/en-us/dotnet/api/system.string.create?view=net-9.0#system-string-create-1(system-int32-0-system-buffers-spanaction((system-char-0)))

Reply:
Will update the other one.